client/components/course/Sidebar/Body.vue

```diff
@@ -70,9 +114,33 @@ export default {
       const data = { ...this.activity.data, [key]: value };
       this.update({ _cid: this.activity._cid, data });
     },
-    publishActivity() {
+    publishData(activities = [this.activity]) {
       this.publishing = true;
-      this.publish(this.activity).then(() => (this.publishing = false));
+      Promise.each(activities, activity => {
+        this.publishStatus = `Publishing ${activity.data.name}`;
+        return (this.publish(activity));
+      }).then(() => {
+        this.publishing = false;
+        this.publishStatus = '';
+      });
+    },
+    publishConfirmation(activities, type) {
+      const message = this.getMessage(type);
+      appChannel.emit('showConfirmationModal', {
+        type: 'publish',
+        message: message,
+        action: () => {
+          this.publishData(activities);
+        }
+      });
+    },
+    getMessage(type) {
+      if (type === 'publishAll') {
```

Comment:
Ternary expression that checks passed activities length vs outline activities length.